### PR TITLE
Issues/#26

### PR DESCRIPTION
**Pull request by hg-pyun**
#26 Feature development.
- Type checking the iterator parameter is RepeatIterator
- Refactoring validator
- Please ignore 85cb839, 922664c, 67e720b commits

### CODE REVIEW

**Comment by SungJinYoo** [comment]
Changing the base branch to `issue/#30` for easy reviewing. Correct when merging this PR

File: README.md [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 [![npm](https://img.shields.io/npm/v/iterize.svg)](https://www.npmjs.com/package/iterize)
-![npm](https://img.shields.io/npm/dt/iterize.svg)
+[![npm](https://img.shields.io/npm/dt/iterize.svg)](https://www.npmjs.com/package/iterize)
 [![GitHub license](https://img.shields.io/github/license/hg-pyun/iterize.svg)](https://github.com/hg-pyun/iterize/blob/master/LICENSE)
 
-> Minimalistic JavaScript Iterators
+> Use JavaScript Iterator, Easily 
 
-A great feature called Iterator was added into JavaScript. However, it's a strange concept to most of the front-end developers.
+**iterize** is minimalistic creator for iterator. A great feature called Iterator was added into JavaScript. However, it's a strange concept to most of the front-end developers.
```
Comment (SungJinYoo) [inline README.md]:
```suggestion
**iterize** is a minimalistic creator for the iterator. A great feature called Iterator was added into JavaScript. However, it's a strange concept for most of the front-end developers.
```

File: README.md [hunk 2]
```diff
@@ -97,7 +107,7 @@ iter.next(); // { value: 1, done: false }
 iter.next(); // { value: 2, done: false }
 ...
 ```
-
+`Cycle` can also receive range iterator.
```
Comment (SungJinYoo) [inline README.md]:
```suggestion
`Cycle` can also receive a range iterator.
```

File: README.md [hunk 3]
```diff
@@ -52,7 +56,23 @@ range(start: number, end?:number, step?: number | Function): Iterator
...
 
+for (let number of range(0, 5)) {
+    console.log(number);  // 0, 1, 2, 3, 4
+}
+
+for (let number of range(0, 5, 2)) {
+    console.log(number);  // 0, 2, 4
+}
+```
+With spread operator.
```
Comment (SungJinYoo) [inline README.md]:
```suggestion
With the spread operator.
```

File: README.md [hunk 4]
```diff
@@ -156,24 +166,20 @@ replicate(count: number, item: number | string | Iterator): Iterator
...
 import {replicate} from 'iterize';
 
-[...replicate(5, 0)]   // [0, 0, 0, 0, 0]
-[...replicate(5, 'a')] // ['a', 'a', 'a', 'a', 'a']
+for (let number of replicate(3, 1)) {
+    console.log(number);  // [1, 1, 1]
+}
 ```
-
+With spread operator.
```
Comment (SungJinYoo) [inline README.md]:
```suggestion
With the spread operator.
```

File: README.md [hunk 5]
```diff
@@ -188,27 +194,24 @@ Returns the first N items of the iterator sequentially.
...
 import {take, cycle} from 'iterize';
 
 const cycleIterator = cycle([1, 2, 3]);
-[...take(5, cycleIterator)]; // [1, 2, 3, 1, 2]
+for (let number of take(5, cycleIterator)) {
+    console.log(number);  // 1, 2, 3, 1, 2
+}
 ```
-
+With spread operator.
```
Comment (SungJinYoo) [inline README.md]:
```suggestion
With the spread operator.
```

File: README.md (hunk 1 above)
Comment (hg-pyun) [reply README.md]: Thank you!

**Pull request merged**